# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/DecoderTypes.h
# Commit: b1577155be1c
# Full Hash: b1577155be1c9e9070c91a969c0cdcfaab9c8936
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - harzar fix. CLOSED TREE
#   
#   
#   This patch uses RootedDictionary around `EncodedVideoChunkMetadata`,
#   `VideoDecoderConfig` and `VideoColorSpaceInit` in `OutputEncodedData` to
# ==============================================================================

diff -r ecf1c8556163 -r b1577155be1c dom/media/webcodecs/DecoderTypes.h
--- a/dom/media/webcodecs/DecoderTypes.h	Sat Dec 23 10:32:48 2023 +0200
+++ b/dom/media/webcodecs/DecoderTypes.h	Sat Dec 23 08:17:11 2023 +0000
@@ -43,6 +43,15 @@
  public:
   static UniquePtr<VideoDecoderConfigInternal> Create(
       const VideoDecoderConfig& aConfig);
+  VideoDecoderConfigInternal(const nsAString& aCodec,
+                             Maybe<uint32_t>&& aCodedHeight,
+                             Maybe<uint32_t>&& aCodedWidth,
+                             Maybe<VideoColorSpaceInternal>&& aColorSpace,
+                             Maybe<RefPtr<MediaByteBuffer>>&& aDescription,
+                             Maybe<uint32_t>&& aDisplayAspectHeight,
+                             Maybe<uint32_t>&& aDisplayAspectWidth,
+                             const HardwareAcceleration& aHardwareAcceleration,
+                             Maybe<bool>&& aOptimizeForLatency);
   ~VideoDecoderConfigInternal() = default;
 
   nsString ToString() const;
@@ -79,17 +88,6 @@
   Maybe<uint32_t> mDisplayAspectWidth;
   HardwareAcceleration mHardwareAcceleration;
   Maybe<bool> mOptimizeForLatency;
-
- private:
-  VideoDecoderConfigInternal(const nsAString& aCodec,
-                             Maybe<uint32_t>&& aCodedHeight,
-                             Maybe<uint32_t>&& aCodedWidth,
-                             Maybe<VideoColorSpaceInternal>&& aColorSpace,
-                             Maybe<RefPtr<MediaByteBuffer>>&& aDescription,
-                             Maybe<uint32_t>&& aDisplayAspectHeight,
-                             Maybe<uint32_t>&& aDisplayAspectWidth,
-                             const HardwareAcceleration& aHardwareAcceleration,
-                             Maybe<bool>&& aOptimizeForLatency);
 };
 
 class VideoDecoderTraits {